azure/src/main/java/sunstone/azure/impl/AzureArmTemplateCloudDeploymentManager.java

```diff
@@ -60,12 +64,35 @@ void deploy(String template, Map<String, String> parameters, String group, Strin
             LOGGER.warn("Azure resource group '{}' already exists! It will be reused and deleted when tests are finished.", group);
         }
 
-        armManager.deployments().define(deploymentName)
+        //.create() doesn't allow status check and fails
+        Accepted<Deployment> acceptedDeployment = armManager.deployments().define(deploymentName)
                 .withExistingResourceGroup(group)
                 .withTemplate(template)
                 .withParameters(parametersFromMap(template, parameters))
                 .withMode(DeploymentMode.INCREMENTAL)
-                .create();
+                .beginCreate();
+
+        // polling wait
+        final Duration pollInterval = Duration.ofSeconds(10);
+        LongRunningOperationStatus pollStatus = acceptedDeployment.getActivationResponse().getStatus();
+        long delayInMills = pollInterval.toMillis();
+        while (!pollStatus.isComplete()) {
+            try {
+                Thread.sleep(delayInMills);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+
+            PollResponse<?> pollResponse = acceptedDeployment.getSyncPoller().poll();
+            pollStatus = pollResponse.getStatus();
+        }
+        if (pollStatus != LongRunningOperationStatus.SUCCESSFULLY_COMPLETED) {
+            LOGGER.error("Azure deployment from template {} in \"{}\" group failed", deploymentName, group);
+            AzureUtils.downloadResourceGroupLogs(armManager, group);
+            undeploy(group);
```

Comment:
Can the undeploy be controlled by a sunstone property? I have found it useful being able to investigate the resources after the deployment failed.

Reply:
I would tackle this in separate PR changing undeploy(..) logic for #201